core/domain/question_services.py

```diff
@@ -217,3 +228,319 @@ def update_question(
+
+
+def get_summaries_of_linked_skills(question_id):
+    """Gets linked skill IDs for given question.
+
+    Args:
+        question_id: str. The question ID for the given question.
+
+    Returns:
+        QuestionSkillLinkModel. The QuestionSkillModel for the given question.
+    """
+    linked_skill_summaries = []
+    question_skill_links = question_models.QuestionSkillLinkModel.get(
+        question_id, strict=False)
+
+    if question_skill_links is None:
+        return None
+    for question_skill_link in question_skill_links:
+        linked_skill_summaries = (
+            skill_models.SkillSummaryModel.get(
+                question_skill_link.skill_id, strict=False))
+
+    return linked_skill_summaries
+
+
+def get_question_rights_from_model(question_rights_model):
+    """Constructs a QuestionRights object from the given question rights model.
+
+    Args:
+        question_rights_model: QuestionRightsModel. Question rights from the
+            datastore.
+
+    Returns:
+        QuestionRights. The rights object created from the model.
+    """
+
+    return question_domain.QuestionRights(
+        question_rights_model.id,
+        question_rights_model.manager_ids
+    )
+
+
+def save_question_rights(
+        question_rights, committer_id, commit_message, commit_cmds):
+    """Saves a QuestionRights domain object to the datastore.
+
+    Args:
+        question_rights: QuestionRights. The rights object for the given
+            question.
+        committer_id: str. ID of the committer.
+        commit_message: str. Descriptive message for the commit.
+        commit_cmds: list(QuestionChangeDict). A list of commands
+            describing what kind of commit was done.
+    """
+
+    model = question_models.QuestionRightsModel.get(
+        question_rights.id, strict=False)
+
+    model.manager_ids = question_rights.manager_ids
+    commit_cmd_dicts = [commit_cmd.to_dict() for commit_cmd in commit_cmds]
+    model.commit(committer_id, commit_message, commit_cmd_dicts)
+
+
+def create_new_question_rights(question_id, committer_id):
+    """Creates a new question rights object and saves it to the datastore.
+
+    Args:
+        question_id: str. ID of the question.
+        committer_id: str. ID of the committer.
+    """
+    question_rights = question_domain.QuestionRights(question_id, [])
+    commit_cmds = [{'cmd': CMD_CREATE_NEW}]
+
+    question_models.QuestionRightsModel(
+        id=question_rights.id,
+        manager_ids=question_rights.manager_ids
+    ).commit(committer_id, 'Created new question rights', commit_cmds)
+
+
+def get_question_rights(question_id, strict=True):
+    """Retrieves the rights object for the given question.
+
+    Args:
+        question_id: str. ID of the question.
+        strict: bool. Whether to fail noisily if no question rights with a
+            given id exists in the datastore.
+
+    Returns:
+        QuestionRights. The rights object associated with the given question.
```

Comment:
Here also, write about the case where None would be returned.

Reply:
done

Reply:
Is it done? Please mark it as done, only after you have actually completed it. I meant to write the None case in the docstring.